src/test/java/com/mongodb/hibernate/jdbc/MongoStatementIntegrationTests.java

```diff
@@ -0,0 +1,349 @@
+            });
+
+            // when && then
+            var deleteMql =
+                    """
+                    {
+                        delete: "%s",
+                        deletes: [
+                            {
+                                q: { author: "Leo Tolstoy" },
+                                limit: 1
+                            }
+                        ]
+                    }"""
+                            .formatted(COLLECTION_NAME);
+            var expectedDocs = Set.of(
+                    BsonDocument.parse(
+                            """
+                            {
+                                    _id: 2,
+                                    title: "Anna Karenina",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                 }"""),
+                    BsonDocument.parse(
+                            """
+                            {
+                                    _id: 3,
+                                    title: "Resurrection",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 4,
+                                    title: "Crime and Punishment",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 5,
+                                    title: "The Brothers Karamazov",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                }"""),
+                    BsonDocument.parse(
+                            """
+                           {
+                                    _id: 6
+                                    title: "Fathers and Sons",
+                                    author: "Ivan Turgenev",
+                                    outOfStock: false
+                                }"""));
+            assertExecuteUpdate(deleteMql, autoCommit, 1, expectedDocs);
+        }
+
+        private void assertExecuteUpdate(
+                String mql, boolean autoCommit, int expectedRowCount, Set<? extends BsonDocument> expectedDocuments) {
+            session.doWork(connection -> {
+                connection.setAutoCommit(autoCommit);
+                var statement = (MongoStatement) connection.createStatement();
+                assertEquals(expectedRowCount, statement.executeUpdate(mql));
+                if (!autoCommit) {
+                    connection.commit();
+                }
```

Comment:
Let's guarantee the commit:

```suggestion
                try {
                    assertEquals(expectedRowCount, statement.executeUpdate(mql));
                } finally {
                    if (!autoCommit) {
                        connection.commit();
                    }
                }
```

Reply:
I am afraid the above change won't work for the commit() has to happen BEFORE assertion statements, otherwise empty result will be returned.